plenum/common/stacks.py

```diff
@@ -34,14 +37,70 @@ def __init__(self, stackParams: dict, msgHandler: Callable, seed=None,
             seed=seed,
             onlyListener=True,
             config=config,
-            msgRejectHandler=msgRejectHandler)
+            msgRejectHandler=msgRejectHandler,
+            create_listener_monitor=True)
         MessageProcessor.__init__(self, allowDictOnly=False)
-        self.connectedClients = set()
+
+        self.max_connected_clients_num = config.MAX_CONNECTED_CLIENTS_NUM
+        self.postrestart_wait_time = config.STACK_POSTRESTART_WAIT_TIME
+        self.min_stack_restart_timeout = config.MIN_STACK_RESTART_TIMEOUT
+        self.max_stack_restart_time_deviation = config.MAX_STACK_RESTART_TIME_DEVIATION
+        self.init_stack_restart_params()
+
+    def init_stack_restart_params(self):
+        self.connected_clients_num = 0
+        self.stack_restart_is_needed = False
+        self.last_start_time = time.time()
```

Comment:
Should we use `time.perf_counter()`?

Reply:
Seems like time.perf_counter() is not appropriate here. From time module reference:
"The reference point of the returned value is undefined, so that only the difference between the results of consecutive calls is valid."
But I'm calculating a time point in future. So using time.time() function is much more understandable for me as it returns the time in seconds since the epoch.
And also about time.perf_counter(): "A clock with the highest available resolution to measure a short duration." We don't need the highest resolution here as calculated time out is about tens of minutes.